src/main/kotlin/exercise3/task3/SherlockValidatesString.kt

```diff
@@ -29,7 +29,24 @@ package exercise3.task3
  */
 
 internal fun isSherlockValid(s: String): String {
-    TODO("Implement me!!!")
+    val charCounts = s.groupBy { it }.values.map { it.size }
+
+    if (charCounts.toSet().size == 1) {
```

Comment:
`charCounts.toSet()` could have be done in the line above where we define `charCounts` so that we avoid using `toSet` multiple times.